frontends/web/src/routes/accounts/select-receive.tsx

```diff
@@ -16,27 +16,33 @@
 
 import { useEffect, useState } from 'react';
 import { useTranslation } from 'react-i18next';
-import { getBalance, IAccount } from '../../api/account';
-import { AccountSelector, TOption } from '../../components/accountselector/accountselector';
+import { IAccount } from '../../api/account';
 import { Header } from '../../components/layout';
 import { route } from '../../utils/route';
-import { isBitcoinOnly } from '../account/utils';
+import { getAccountsByKeystore, isAmbiguiousName, isBitcoinOnly } from '../account/utils';
 import { View, ViewContent } from '../../components/view/view';
+import { GroupedAccountSelector, TGroupedOption, TOption } from '../../components/groupedaccountselector/groupedaccountselector';
+import { getBalancesForGroupedAccountSelector } from '../../components/groupedaccountselector/services';
 
 type TReceiveAccountsSelector = {
     activeAccounts: IAccount[]
 }
 export const ReceiveAccountsSelector = ({ activeAccounts }: TReceiveAccountsSelector) => {
-  const [options, setOptions] = useState<TOption[]>([]);
   const [code, setCode] = useState('');
+  const [groupedOptions, setGroupedOptions] = useState<TGroupedOption[]>();
   const { t } = useTranslation();
 
   useEffect(() => {
-    const options = activeAccounts.map(account => ({ label: account.name, value: account.code, disabled: false, coinCode: account.coinCode } as TOption));
+    const accountNames = getAccountsByKeystore(activeAccounts);
     //setting options without balance
-    setOptions(options);
+    const groupedOpts: TGroupedOption[] = accountNames.map(grouped => ({
+      label: `${grouped.keystore.name} ${isAmbiguiousName(grouped.keystore.name, accountNames) ? `(${grouped.keystore.rootFingerprint})` : ''}`,
+      connected: grouped.keystore.connected,
+      options: grouped.accounts.map((account) => ({ label: account.name, value: account.code, coinCode: account.coinCode, disabled: false })) as TOption[]
+    }));
```

Comment:
how about destructing accounts and keystores?

```ts
    const groupedOpts: TGroupedOption[] = accountNames.map(({ accounts, keystore }) => ({
      label: `${keystore.name} ${isAmbiguiousName(keystore.name, accountNames) ? `(${keystore.rootFingerprint})` : ''}`,
      connected: keystore.connected,
      options: accounts.map((account) => ({ label: account.name, value: account.code, coinCode: account.coinCode, disabled: false })) as TOption[]
    }));
```

Reply:
tbh I prefer it this way. To me it's clearer which `name` it is for example if we explicitly do `keystore.name` than just `name` knowing in this block we also have `account.name`. Also clearer for me personally to see that `rootFingerPrint` for ex, comes from keystore by doing`keystore.`. 

 But if you strongly prefer destructured, we could do that too ofc :)